sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.cs

```diff
@@ -528,27 +528,54 @@ public static void RetryOnException(int retries)
             } while (true);
         }
 
+        public static void ComplexRetryOnException(int retries)
+        {
+            var attempts = 0;
+            try
+            {
+                do
+                {
+                    if (retries > 0)
+                    {
+                        DoNothing();
+                        try
+                        {
+                            attempts++;
+                            DoNothing();
+                            break;
+                        }
+                        catch (ArgumentException ex)
+                        {
+                            DoNothing();
+                        }
+                    }
+                } while (true);
+            }
+            catch (Exception ex)
+            {
+                if (attempts > retries)
+                    throw;
+                DoNothing();
+            }
+        }
+
         public void Bar()
         {
             bool isFirst = true;
             foreach (var i in System.Linq.Enumerable.Range(1, 10))
             {
                 try
                 {
-                    if (isFirst)
```

Comment:
this is just a simplification of the test reported in #2393 (it did not affect the reproducibility of the FP)